### PR TITLE
Fixes #9252: System variable "MANAGED_NODES_KEY" is empty

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/9252

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/rudder/domain/nodes/NodeInfo.scala [hunk 1]
```diff
@@ -157,11 +157,13 @@ object CFEngineKey {
...
    * Oh, and Java Security API force a lot of casting.
    * This is insane.
+   *
+   * Caution: to use the complete key, with header and footer, you need to use key.key
    */
   def getHash(key: PublicKey): Box[String] = {
 
     //the parser able to read PEM files
-    val parser = new PEMParser(new StringReader(key.value))
+    val parser = new PEMParser(new StringReader(key.key))
```
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/domain/nodes/NodeInfo.scala]: Shouldn't we also add a test on null for parser with something like:
`
for {
   parser <- Box(Option(new PEMParser(new StringReader(key.key))))
...
`
 Or was the NPE elsewhere ?

**Comment by ncharles** [comment]
Commit modified
